### PR TITLE
Trust Apple's status for validity check

### LINKED ISSUE #6137: [Sigh] recreating provision profiles even with valid certificate
### New Issue Checklist
- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
##### Complete output when running fastlane, including the stack trace and command used

```
sigh -a [app id] -b [team id] -u [username]

[15:10:16]: Starting login with user [username]
[15:10:18]: Successfully logged in
[15:10:18]: Fetching profiles...
[15:10:19]: No existing profiles found, that match the certificates you have installed locally! Creating a new provisioning profile for you
[15:10:19]: Found more than one code signing identity. Choosing the first one. Check out `sigh --help` to see all available options.
[15:10:19]: Available Code Signing Identities for current filters:
[15:10:19]:     - Name: [REDACTED]
[15:10:19]:     - Name: [REDACTED]
[15:10:20]: The name '[REDACTED] InHouse' is already taken, using another one.
[15:10:20]: Creating new provisioning profile for [REDACTED] with name '[REDACTED] InHouse 1473966620'
[15:10:21]: Downloading provisioning profile...
[15:10:22]: Successfully downloaded provisioning profile...
[15:10:22]: Installing provisioning profile...
[REDACTED]
[15:10:22]: Setting Provisioning Profile type to 'enterprise'
```

sigh version (run `fastlane -v`): 1.10.2

Do you use bundler to execute fastlane (i.e. `bundle exec fastlane`)? yes

Do you use a Ruby environment manager (e.g. `chruby`, `rbenv`, `rvm`)? rvm

Additionally, when debugging sigh I see the following in the profiles

```
[<Spaceship::Portal::ProvisioningProfile::InHouse 
    id="[REDACTED]", 
    uuid="[REDACTED]", 
    expires=#<DateTime: 2017-09-15T19:12:19+00:00 ((2458012j,69139s,0n),+0s,2299161j)>, 
    distribution_method="inhouse", 
    name="[REDACTED] InHouse 1473966738", 
    status="Active", 
    type="iOS UniversalDistribution", 
    version="2", 
    platform="ios", 
    managing_app=nil, 
    app=<Spaceship::Portal::App 
        app_id="[REDACTED]", 
        name="[REDACTED]", 
        platform="ios", 
        prefix="[REDACTED]", 
        bundle_id="[REDACTED]", 
        is_wildcard=false, 
        features={"push"=>true, "iCloud"=>false, "inAppPurchase"=>true, "gameCenter"=>true, "LPLF93JG7M"=>false, "passbook"=>false, "IAD53UNK2F"=>false, "V66P55NK2I"=>false, "dataProtection"=>"untilfirstauth", "SKC3T5S89Y"=>false, "APG3427HIY"=>false, "HK421J6T7P"=>false, "homeKit"=>false, "WC421J6T7P"=>false, "cloudKitVersion"=>1}, 
        enabled_features=["gameCenter", "dataProtection", "inAppPurchase", "push"], 
        dev_push_enabled=false, 
        prod_push_enabled=false, 
        app_groups_count=nil, 
        cloud_containers_count=nil, 
        identifiers_count=nil>, 
    certificates=[], 
    devices=[]>
```

As you can see the certificates is empty as well as devices.

When I login to the portal, the certificate is set and valid with devices.  Also I have the valid certificate downloaded on my machine.

### PR DESCRIPTION
**Pull request by maschall**
Resolves #6137 if using --skip_certificate_verification

This PR moves the certificate_valid? check to after the check for skip_certificate_verification.

### CODE REVIEW

File: spaceship/lib/spaceship/portal/provisioning_profile.rb [hunk 1]
```diff
@@ -379,7 +379,7 @@ def certificate_valid?
 
       # @return (Bool) Is the current provisioning profile valid?
       def valid?
```
Comment (maschall) [inline spaceship/lib/spaceship/portal/provisioning_profile.rb]: from doing a find all on the project, this method is only called in runner.sh and the specs.  So it shouldn't effect any other systems.

**Comment by KrauseFx** [comment]
Why does moving the method call solve the problem?

**Comment by maschall** [comment]
It doesn't solve the problem entirely, however it does more accurately follow the flag --skip_certificate_verification, by not verifying the certificates if the flag is true.

**Comment by maschall** [comment]
Closing as the original issue was fixed

**Comment by KrauseFx** [comment]
Yeah 👍

**Pull request closed without merging**